### PR TITLE
Adding SNI and remote server validation.

### PR DESCRIPTION
**Pull request by CIPop**
Adding SNI and remote server certificate validation.

### CODE REVIEW

File: AzureIotPnpDps.X/platform/cloud/cloud_service.c [hunk 1]
```diff
@@ -221,8 +221,19 @@ static int8_t connectMQTTSocket(void)
...
+			int optVal = 1;
+
+			ret = BSD_setsockopt(*context->tcpClientSocket, SOL_SSL_SOCKET, SO_SSL_SNI, mqtt_host, strlen(mqtt_host));
+			if (ret == BSD_SUCCESS) {
+				ret = BSD_setsockopt(*context->tcpClientSocket, SOL_SSL_SOCKET, SO_SSL_ENABLE_SNI_VALIDATION, &optVal, sizeof(optVal));
+				if (ret == BSD_SUCCESS)
+				{
+			        debug_print("CLOUD: Connect socket");
+			        ret = BSD_connect(*context->tcpClientSocket, (struct bsd_sockaddr*) & addr, sizeof (struct bsd_sockaddr_in));
+			    }
```
Comment (danewalton-msft) [inline AzureIotPnpDps.X/platform/cloud/cloud_service.c]: nit: can't tell if this is Github but spacing here.

**Review by danewalton-msft — APPROVED** [review]

File: AzureIotPnpDps.X/platform/cloud/cloud_service.c (hunk 1 above)
Comment (CIPop) [reply AzureIotPnpDps.X/platform/cloud/cloud_service.c]: Thanks! It mixed spaces with tabs. I'll convert back to tab characters.

File: AzureIotPnpDps.X/platform/cloud/cloud_service.c [hunk 2]
```diff
@@ -221,13 +221,24 @@ static int8_t connectMQTTSocket(void)
 
 		socketState = BSD_GetSocketState(*context->tcpClientSocket);
 		if (socketState == SOCKET_CLOSED) {
-			debug_print("CLOUD: Connect socket");
-			ret = BSD_connect(*context->tcpClientSocket, (struct bsd_sockaddr*) & addr, sizeof(struct bsd_sockaddr_in));
+			debug_print("Configuring SNI to connect to %s", mqtt_host);
+
+			int optVal = 1;
```
Comment (ericwolz) [inline AzureIotPnpDps.X/platform/cloud/cloud_service.c]: move to line 230?

File: AzureIotPnpDps.X/platform/cloud/cloud_service.c [hunk 3]
```diff
@@ -221,13 +221,24 @@ static int8_t connectMQTTSocket(void)
...
+
+			ret = BSD_setsockopt(*context->tcpClientSocket, SOL_SSL_SOCKET, SO_SSL_SNI, mqtt_host, strlen(mqtt_host));
+			if (ret == BSD_SUCCESS) {
+				ret = BSD_setsockopt(*context->tcpClientSocket, SOL_SSL_SOCKET, SO_SSL_ENABLE_SNI_VALIDATION, &optVal, sizeof(optVal));
+				if (ret == BSD_SUCCESS)
```
Comment (ericwolz) [inline AzureIotPnpDps.X/platform/cloud/cloud_service.c]: you can remove the nesting and put it outside of this if statement

**Review by ericwolz — APPROVED** [review]

**Comment by CIPop** [comment]
@ericwol-msft I have addressed the comments and ran an e2e validation on the device.

**Pull request merged**